src/main/java/org/prebid/server/bidder/eplanning/EplanningBidder.java

```diff
@@ -193,19 +195,45 @@ private String resolveRequestUri(BidRequest request, List<String> requestsString
         final String pageUrl = site != null && StringUtils.isNotBlank(site.getPage())
                 ? site.getPage() : DEFAULT_PAGE_URL;
 
-        String uri = endpointUrl + String.format("/%s/%s/%s/%s?ct=1&r=pbs&ncb=1&ur=%s&e=%s",
-                clientId, DFP_CLIENT_ID, pageDomain, SEC, HttpUtil.encodeUrl(pageUrl),
-                String.join("+", requestsStrings));
+        final String requestTarget = request.getApp() != null && StringUtils.isNotBlank(request.getApp().getBundle())
+                ? request.getApp().getBundle()
+                : pageDomain;
+
+        final String uri = endpointUrl + String.format("/%s/%s/%s/%s", clientId, DFP_CLIENT_ID, requestTarget, SEC);
+
+        final URIBuilder uriBuilder = new URIBuilder()
+                .setPath(uri)
+                .addParameter("r", "pbs")
+                .addParameter("ncb", "1");
+
+        if (request.getApp() == null) {
+            uriBuilder.addParameter("ur", pageUrl);
+        }
+        uriBuilder.addParameter("e", String.join("+", requestsStrings));
 
         final User user = request.getUser();
         if (user != null && StringUtils.isNotBlank(user.getBuyeruid())) {
```

Comment:
Would it be better if we get a small refactoring for fetching the value like:
`final String buyeruid = user != null ? user.getBuyeruid() : null;`

then just replace current condition with `buyeruid != null`.

But up to you.